minecraft/listener.go

```diff
@@ -227,14 +254,17 @@ func (listener *Listener) listen() {
 // createConn creates a connection for the net.Conn passed and adds it to the listener, so that it may be
 // accepted once its login sequence is complete.
 func (listener *Listener) createConn(netConn net.Conn) {
+	listener.packsMu.RLock()
+	defer listener.packsMu.RUnlock()
```

Comment:
This might not be safe here because `listener.packs` could be changed and read simultaneously (as it's a reference type). Might want to RLock, then `packs := slices.Clone(listener.packs)`, then unlock and pass `packs` to `conn.resourcePacks`.